### DISCUSSION TITLE
How to confirm pop up appears?

### DISCUSSION THREAD

**Discussion by Harsha-Gelato** [appium/appium#17317] [General]
Hi All,

I am currently automating one testCase for IOS, in which when I click on element to rate the app, it opens up the rating pop up, in which I am trying to confirm:

whether the pop up appears on clicking on the “Rate the app” element
Click on “Not now” to remove the pop up from the screen.
Does anyone knows how can I achieve these two task in IOS.
<img width="330" alt="Screenshot 2022-08-04 at 10 24 47" src="https://user-images.githubusercontent.com/103107117/182894046-5125a067-0cd5-4731-8977-21c4737fdf5c.png">

**Comment by KazuCocoa** [comment]
https://discuss.appium.io/ may help to ask such questions.
Basically, you can confirm the element with find_element/s with accessibility_id, predicate or class chain, and click the "not now".

**Reply by Harsha-Gelato** [reply]
I added this question there as well https://discuss.appium.io/t/how-to-confirm-pop-up-appears/36980 but did not get any response.
<img width="1281" alt="Screenshot 2022-08-04 at 10 12 04" src="https://user-images.githubusercontent.com/103107117/183031915-39eae69b-b5e5-4bd5-84e8-fe09e7ab478f.png">

The attributes listed eg xpath seems to be very big and complex. What is your suggestions in this type of case?

**Reply by KazuCocoa** [reply]
I see.

I usually use accessibility_id (name value) if the word is unique in the element tree, then predict (or class chain). XPath is last choice since usually XPath is the slowest in iOS (since XCTest framework does not provide it natively.)
https://appiumpro.com/editions/60-how-to-pick-the-right-locator-strategy
